static/src/javascripts-legacy/projects/common/modules/atoms/story-questions.js

```diff
@@ -19,111 +19,98 @@ define([
                 mode: 'cors',
                 method: 'POST',
                 body: {email : email.value, listId: listId.value}
             })
             .then(function (response) {
                 if (response.ok) {
-                    var submissionContainerEl = answersEmailSignupForm.parentElement;
-                    var thankyouMessage = document.getElementById('js-final-thankyou-message-' + questionId);
-
-                    if (submissionContainerEl && thankyouMessage) {
-                        submissionContainerEl.classList.add('is-hidden');
-                        thankyouMessage.classList.remove('is-hidden');
-                    }
+                    var component = event.currentTarget.closest('.js-view-tracking-component');
+                    $('.js-storyquestion-email-done', component).removeClass('is-hidden');
+                    $('.js-storyquestion-email-question', component).addClass('is-hidden');
+                    $('.js-storyquestion-email-signup-form', component).addClass('is-hidden');
+                    $('.js-storyquestion-email-enter', component).addClass('is-hidden');
+                    $('.js-storyquestion-email-nospam', component).addClass('is-hidden');
                 }
             });
         }
     }
 
     return {
         init: function() {
-            var askQuestionLinks = $('.js-ask-question-link');
-            var isEmailSubmissionReadyElement = document.getElementById('js-storyquestion-is-email-submission-ready');
-
-            var isEmailSubmissionReady = false;
-
-            if (isEmailSubmissionReadyElement) {
-                isEmailSubmissionReady = isEmailSubmissionReadyElement.dataset.isEmailSubmissionReady ? isEmailSubmissionReadyElement.dataset.isEmailSubmissionReady : false;
-            }
 
-            askQuestionLinks.each(function (el) {
-                bean.on(el, 'click', function(event) {
-                    askQuestion(event, isEmailSubmissionReady)
-                    this.classList.add('is-clicked')
-                });
+            $('.js-storyquestion-email-question-button').each(function(el) {
+                bean.on(el, 'click', emailMe);
             });
 
-            var answersEmailSignupForms = $('.js-storyquestion-email-signup-form');
-
-            answersEmailSignupForms.each(function (el) {
+            $('.js-storyquestion-email-signup-form').each(function(el) {
                 bean.on(el, 'submit', submitSignUpForm);
             });
 
-
             Id.getUserFromApi(function (userFromId) {
-                if (userFromId && userFromId.primaryEmailAddress) {
+                var signedIn = (userFromId && userFromId.primaryEmailAddress) != null;
+
+                if (signedIn) {
                     fastdom.write(function () {
                         $('.js-storyquestion-email-signup-form').each(function(form) {
                             $('.js-storyquestion-email-signup-button', form).removeClass('button--with-input');
```

Comment:
Little trick here, you can write `$('.button--with-input', form).removeClass('button--with-input');` instead.